test/spec/modules/33acrossAnalyticsAdapter_spec.js

```diff
@@ -0,0 +1,2571 @@
+          const incompleteAnalyticsReport = getStandardAnalyticsReport();
+          incompleteAnalyticsReport.auctions.forEach(auction => {
+            auction.adUnits.forEach(adUnit => {
+              adUnit.bids.forEach(bid => {
+                bid.hasWon = 0;
+              });
+            });
+          });
+
+          navigator.sendBeacon.returns(true);
+
+          const endpoint = faker.internet.url();
+          this.enableAnalytics({ endpoint });
+
+          performStandardAuction({exclude: ['bidWon', 'auctionEnd']});
+          sandbox.clock.tick(this.defaultTimeout + 1);
+
+          assert.calledOnceWithStringJsonEquivalent(navigator.sendBeacon, endpoint, incompleteAnalyticsReport);
+        });
+
+        it('it logs an info message', function () {
+          const incompleteAnalyticsReport = getStandardAnalyticsReport();
+          incompleteAnalyticsReport.auctions.forEach(auction => {
+            auction.adUnits.forEach(adUnit => {
+              adUnit.bids.forEach(bid => {
+                bid.hasWon = 0;
+              });
+            });
+          });
+
+          navigator.sendBeacon.returns(true);
+
+          const endpoint = faker.internet.url();
+          this.enableAnalytics({ endpoint });
+
+          performStandardAuction({exclude: ['bidWon', 'auctionEnd']});
+          sandbox.clock.tick(this.defaultTimeout + 1);
+
+          assert.calledWithExactly(log.info, `Analytics report sent to ${endpoint}`, incompleteAnalyticsReport);
+        });
+      });
+    });
+  });
+});
+
+function performStandardAuction({ exclude = [] } = {}) {
+  const mockEvents = getMockEvents();
+  const { prebid, gam } = mockEvents;
+  const [auction] = prebid;
+
+  if (!exclude.includes(EVENTS.AUCTION_INIT)) {
+    events.emit(EVENTS.AUCTION_INIT, auction.AUCTION_INIT);
+  }
+
+  if (!exclude.includes(EVENTS.BID_REQUESTED)) {
+    for (let bidRequestedEvent of auction.BID_REQUESTED) {
+      events.emit(EVENTS.BID_REQUESTED, bidRequestedEvent);
+    };
+  }
+
+  if (!exclude.includes(EVENTS.BID_RESPONSE)) {
+    for (let bidResponseEvent of auction.BID_RESPONSE) {
+      events.emit(EVENTS.BID_RESPONSE, bidResponseEvent);
+    };
+  }
+
+  if (!exclude.includes('slotRenderEnded')) {
+    for (let gEvent of gam.slotRenderEnded) {
+      mockGpt.emitEvent(gEvent);
+    }
+  }
+
+  // Note: AUCTION_END *frequently* (but not always) emits before BID_WON,
+  // even if the auction is ending because all bids have been won.
+  if (!exclude.includes(EVENTS.AUCTION_END)) {
+    events.emit(EVENTS.AUCTION_END, auction.AUCTION_END);
+  }
+
+  if (!exclude.includes(EVENTS.BID_WON)) {
+    for (let bidWonEvent of auction.BID_WON) {
+      events.emit(EVENTS.BID_WON, bidWonEvent);
+    };
+  }
+}
+
+function mapToBids(auctions) {
+  return auctions.flatMap(
+    auction => auction.adUnits.flatMap(
+      au => au.bids
+    )
+  );
+}
+
+function getLocalAssert() {
+  // Derived from ReportDefnition.cue -> AnalyticsReport
```

Comment:
There's a typo there, but it was meant as a hint to the source of truth for the assertion, as defined in our internal docs. I'll remove it since it can quickly become outdated and unhelpful